### PR TITLE
Provide additional context for authorization and logout button

### LINKED ISSUE #8998: Improve accessibility for Authorization buttons
### Content & configuration
The `Authorize` and `Logout` buttons do not provide additional descriptions that are used for improving accessibility, e.g. by screen readers. 

### Is your feature request related to a problem?
N.A.: not an existing filed problem.

### Describe the solution you'd like
Based upon Web Accessibility Guidelines v2.0 (https://www.w3.org/TR/WCAG20/) adding a hidden label would be sufficient to provide addition descriptions without changing the UI itself.

### Describe alternatives you've considered
See https://www.w3.org/WAI/WCAG21/Techniques/html/H44 for the alternatives besides a hidden label (such as roles).

### Additional context
SwaggerAuthorizeButton

### PR DESCRIPTION
**Pull request by vmware-jaret**
### Description
- Introduced the AuthorizeButton component that entails the apply authorization and remove authorization functionality, as well as a close button.
- Unit tests provided for different states.
- AuthorizeButton is able to be used for both form submit and onClick function behavior.

### Motivation and Context
Fixes #8998, mainly to provide additional context in what the user can do by clicking 'Autorize' and 'Remove Authorization'.

### How Has This Been Tested?
Used the swagger-react in another product and have seen that the hidden labels are added. Unittest added + Cypress test fixed.

### Screenshots (if appropriate):
Screenshot 2023-07-05 at 15 22 21

### My PR contains... 
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by vmware-jaret** [comment]
@char0n Conscious of your limited time reviewing all the PRs and reported issues, would be nice to get some feedback both about the reported issue/enhancement and if the PR content makes sense.

**Comment by vmware-jaret** [comment]
@tim-lai Hello, could you review this PR or let me know how to expedite any review / comments?

**Comment by char0n** [comment]
Hi @vmware-jaret,

Thank you for the PR! Your changeset is on my radar, and I'll address it ASAP.

**Comment by vmware-jaret** [comment]
Thanks @char0n. After some analysis indeed the `<label />` element is not suitable for `<input type='submit' />` or `<button />` elements, hence changing to `aria-label` and removed the `AuthorizationButton` type all together.

File: src/core/components/auth/auths.jsx [hunk 1]
```diff
@@ -88,8 +88,8 @@ export default class Auths extends React.Component {
             }
             <div className="auth-btn-wrapper">
               {
-                nonOauthDefinitions.size === authorizedAuth.size ? <Button className="btn modal-btn auth" onClick={ this.logoutClick }>Logout</Button>
-              : <Button type="submit" className="btn modal-btn auth authorize">Authorize</Button>
+                nonOauthDefinitions.size === authorizedAuth.size ? <Button className="btn modal-btn auth" onClick={ this.logoutClick } aria-label="Remove authorization">Logout</Button>
+              : <Button type="submit" className="btn modal-btn auth authorize" aria-label="Apply authorization credentials">Authorize</Button>
```
Comment (char0n) [inline src/core/components/auth/auths.jsx:92]:
```suggestion
              : <Button type="submit" className="btn modal-btn auth authorize" aria-label="Apply credentials">Authorize</Button>
```

File: test/e2e-cypress/e2e/security/apikey.cy.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+describe("ApiKey Authorization", function() {
+  it("should have generic description for authorization button", () => {
+    cy
+      .visit("/?url=/documents/petstore.swagger.yaml")
+      .get(".btn.authorize") // open authorization dialog
+      .click()
+      .get(".modal-ux-content > :nth-child(2)") // only deal with api_key for this test
+      .within(() => {
+        cy.get(".auth-btn-wrapper .authorize")
+          .should("have.attr", "aria-label", "Apply authorization credentials")
```
Comment (char0n) [inline test/e2e-cypress/e2e/security/apikey.cy.js:10]:
```suggestion
          .should("have.attr", "aria-label", "Apply credentials")
```

**Review by char0n — APPROVED** [review]
LGTM!

**Comment by char0n** [comment]
Hi @vmware-jaret,

I've changed `Logout` label slightly to more match the context of the page. The PR is approved, and thank you again for opening this topic.

**Pull request merged**